### PR TITLE
[DOCS] Updated memory settings for Windows

### PR DESCRIPTION
**Pull request by gmarz**
Minor changes to make the memory settings section a little more OS agnostic since `mlockall` now applies to Windows.

Also, added some info on disabling the page file on Windows.

### CODE REVIEW

File: docs/reference/setup/configuration.asciidoc [hunk 1]
```diff
@@ -70,7 +70,7 @@ NOTE: If you installed Elasticsearch using a package (.deb, .rpm) this setting w
 [[setup-configuration-memory]]
 ==== Memory Settings
 
-The Linux kernel tries to use as much memory as possible for file system
+Most operating systems try to use as much memory as possible for file system
 caches and eagerly swaps out unused application memory, possibly resulting
```
Comment (clintongormley) [inline docs/reference/setup/configuration.asciidoc]: swaps -> swap

File: docs/reference/setup/configuration.asciidoc [hunk 2]
```diff
@@ -84,10 +84,16 @@ There are three options:
...
-to have swap enabled.  On Linux systems, you can disable swap temporarily
+to have swap enabled.
+
+On Linux systems, you can disable swap temporarily
 by running: `sudo swapoff -a`. To disable it permanently, you will need
 to edit the `/etc/fstab` file and comment out any lines that contain the
 word `swap`.
+
+On Windows, the equivalent can be achieved by disabling the paging file entirely
+via `System Properties -> Advanced -> Performance -> Advanced -> Virtual memory`.
```
Comment (clintongormley) [inline docs/reference/setup/configuration.asciidoc]: "->"  -> "→"

File: docs/reference/setup/configuration.asciidoc [hunk 3]
```diff
@@ -107,9 +113,9 @@ emergencies.
 * **`mlockall`**
 +
 --
-The third option on Linux/Unix systems only, is to use
-http://opengroup.org/onlinepubs/007908799/xsh/mlockall.html[mlockall] to
-try to lock the process address space into RAM, preventing any Elasticsearch
+The third option is to use
+http://opengroup.org/onlinepubs/007908799/xsh/mlockall.html[mlockall] on Linux/Unix systems, or https://msdn.microsoft.com/en-us/library/windows/desktop/aa366895%28v=vs.85%29.aspx[VirtualLock] on Windows, to
+try and lock the process address space into RAM, preventing any Elasticsearch
```
Comment (clintongormley) [inline docs/reference/setup/configuration.asciidoc]: try and -> try to

File: docs/reference/setup/configuration.asciidoc [hunk 4]
```diff
@@ -360,22 +366,3 @@ the http://logging.apache.org/log4j/1.2/manual.html[log4j documentation].
 Additional Appenders and other logging classes provided by 
 http://logging.apache.org/log4j/extras/[log4j-extras] are also available, 
 out of the box.
-
```
Comment (clintongormley) [inline docs/reference/setup/configuration.asciidoc]: You seem to have deleted the section on deprecation logging?  Presumably you were editing an older version of this file?

Comment (gmarz) [inline docs/reference/setup/configuration.asciidoc]: Ah...yea I originally branched from 1.6 which doesn't have this section.

**Comment by gmarz** [comment]
Thanks for the review @clintongormley !

**Comment by clintongormley** [comment]
LGTM

**Comment by gmarz** [comment]
Merged via https://github.com/elastic/elasticsearch/commit/9b230db095e0a012831d6a040f025526f56588f7

**Pull request closed without merging**